internal/runners/packages/packages.go

```diff
@@ -154,22 +161,80 @@ func executePackageOperation(pj *project.Project, cfg configurable, out output.O
 	return nil
 }
 
+func resolvePkgAndNamespace(prompt prompt.Prompter, packageName string, nsType model.NamespaceType) (string, model.Namespace, error) {
+	ns := model.NewBlankNamespace()
+
+	// Find ingredients that match the input query
+	ingredients, err := model.SearchIngredientsStrict(model.NewBlankNamespace(), packageName, false)
+	if err != nil {
+		return "", ns, locale.WrapError(err, "err_pkgop_search_err", "Failed to check for ingredients.")
+	}
+
+	// If no ingredients matched we give the user an error that provides some alternative suggestions
+	if len(ingredients) == 0 {
+		suggestions, serr := getSuggestions(ns, packageName)
+		if serr != nil {
+			logging.Error("Failed to retrieve suggestions: %v", err)
+		}
+		if len(suggestions) == 0 {
+			return "", ns, locale.WrapInputError(err, "package_ingredient_nomatch", "Could not match {{.V0}}.", packageName)
+		}
+		return "", ns, locale.WrapError(err, "err_pkgop_search",
+			"Could not match {{.V0}}. Did you mean:\n\n{{.V1}}", packageName, strings.Join(suggestions, "\n"))
+	}
+
+	//
+	choices := []string{}
+	values := map[string][]string{}
+	for _, i := range ingredients {
+		language := model.LanguageFromNamespace(*i.Ingredient.PrimaryNamespace)
```

Comment:
While it's unlikely that this value is the empty string, not checking it here can lead to some odd display for the user and invalid namespaces. We could easily check if it's empty and then log and continue.

Reply:
This won't ever be nil, that's a swagger oddity. But it is also guarded against via the `SearchIngredientsStrict` function.